.github/workflows/ci.yml

```diff
@@ -62,11 +62,12 @@ jobs:
         continue-on-error: true
         with:
           path: ${{ matrix.path_ccache }}
-            # TODO: When updating c++ backend, use new date
-          key: ${{ runner.os }}-ccache-${{ matrix.python-version }}-21-02-12
+            # TODO: When updating c++ backend, use new date (yy-mm-dd)
+          key: ${{ runner.os }}-ccache-${{ matrix.python-version }}-21-11-08
 
       - name: Install dependencies
         run: |
+          python -m pip install --upgrade pip setuptools
           # The --upgrade and --upgrade-strategy eager flags ensure that pip will always install the latest allowed version of all the dependencies.
           python -m pip install --upgrade --upgrade-strategy eager -r requirements.txt
```

Comment:
I'm not exactly complaining about this line but since `python -m pip install -e .` will install all the latest dependencies, why do we add this line? `numpy`, `scipy` and `scikit-learn` (which are our dependencies) are not present in the base environment already, so installing them for the first time will also automatically fetch the latest possible version.

Reply:
Looks like we're going for it.  Maybe we can remove this whole step and move the upgrade pip setuptools line to the top of the next step also.